src/signTx.h

```diff
@@ -52,10 +57,18 @@ typedef struct {
 	uint32_t protocolMagic; // part of Byron address
 } common_tx_data_t;
 
+typedef struct {
+	stake_credential_type_t type;
+	union {
+		bip44_path_t pathSpec; // interpretation depends on type // TODO rename to keyPath?
+		uint8_t scriptHash[SCRIPT_HASH_LENGTH];
+	};
+} stake_credential_t;
+
 typedef struct {
 	certificate_type_t type;
-	bip44_path_t pathSpec; // interpretation depends on type // TODO rename to keyPath?
 
+	stake_credential_t stakeCredential;
 	// only for specific types
 	uint8_t poolKeyHash[POOL_KEY_HASH_LENGTH];
 	uint64_t epoch;
```

Comment:
```suggestion
	// certificates are heterogenous, so the fields are relevant only for certain types
	// the grouping into unions is only used to save some memory
	union {
		stake_credential_t stakeCredential;
		uint64_t epoch;
	};
	uint8_t poolKeyHash[POOL_KEY_HASH_LENGTH];
```

Reply:
I also think that adding pool key path to the anonymous union (or somewhere in the union structure where it makes sense) is preferable to storing it in stakeCredential; we might understand what's going on now, but in a couple of months it would look odd.

Reply:
Grouped all the pool retirement data into one struct, in union with the stake credentials, since these two are mutually exclusive.